lib/perf_hooks.js

```diff
@@ -30,6 +33,14 @@ const {
   NODE_PERFORMANCE_MILESTONE_ENVIRONMENT
 } = constants;
 
+const { Buffer } = require('buffer');
+
+const kRunning = Symbol('delay-timer-running');
+const delayTimer = Buffer.alloc(sizeOfDelayTimer);
```

Comment:
This should not be a pooled buffer, because it’s very long-lived … I think `new Uint8Array()` would be fine.

But: The practice storing libuv handles in `Buffer`s in the first place is … erm, a bit unorthodox, and it has some consequences: In particular, with Workers or in embedder scenarios, you will get memory errors if the `Buffer` memory is torn down before the libuv loop, and since the loop typically exists in a larger scope than the V8 `Isolate`, that’s likely the case.

For example, with this PR:

```js
$ ./node --experimental-worker
> void new worker_threads.Worker(`require("perf_hooks").performance.startEventLoopTimer(); process.exit()`, {eval: true}) 
undefined
> .exit
corrupted double-linked list
Aborted (core dumped)
```

You could probably take care of this by registering `Environment` cleanup hooks in the native start/stop functions, but I’d maybe consider a less unusual approach, like treating it like the other native handles in `env.h` (`timer_handle_`, `immediate_check_handle_`, etc.), or using a `HandleWrap`, or maybe looking into a JS-based implementation?

Reply:
how about exposing an AliasedBuffer from c++ land like we do for checking if the event loop should continue spinning?

Reply:
@devsnek That would not solve the bigger issue of storing a libuv handle in JS-managed memory either…

Reply:
@addaleax oh i didn't mean the handle itself should be stored in the aliased buffer, just the two times that are accessed.